db/db.go

```diff
@@ -596,6 +596,20 @@ func GetValidatorDeposits(publicKey []byte) (*types.ValidatorDeposits, error) {
 		deposits.LastEth1DepositTs = deposits.Eth1Deposits[len(deposits.Eth1Deposits)-1].BlockTs
 	}
 
+	// retrieve address names from bigtable
+	names := make(map[string]string)
+	for _, v := range deposits.Eth1Deposits {
+		names[string(v.FromAddress)] = ""
+	}
+	names, _, err = BigtableClient.GetAddressesNamesArMetadata(&names, nil)
```

Comment:
**Question:** Wouldn't it be enough to simply call
https://github.com/gobitfly/eth2-beaconchain-explorer/blob/8c2348627597d5abb97ac16efe506a89f1fda4f1/db/ens.go#L604
here?

I am pretty sure that `GetAddressesNamesArMetadata` is the wrong function. For example, for the zero address on Sepolia it returns "ZERO_ADDRESS" as name. We want to show the ENS though and "ZERO_ADDRESS" is not a valid END name.

If I am right, please also consider removing
https://github.com/gobitfly/eth2-beaconchain-explorer/blob/e92a323f8087281e9b9b908467db7026d85388b1/utils/format.go#L342-L346

We want to show an ENS name like done for example [HERE](https://prater.beaconcha.in/address/0x000002b4A2e3459331b1eB6066A8FdB39c7fdC5D):
![image](https://github.com/gobitfly/eth2-beaconchain-explorer/assets/114066135/d92a6a94-a790-4700-b1b3-b0ae43a50be3)

There should already be a format function. If not it would be good to implement one and to use it wherever ENS names are shown.

Reply:
That screenshot is exactly (one of the spots) where I was getting that function `GetAddressesNamesArMetadata` from (coming from [`AddressTransactionsTableGrid`](https://github.com/gobitfly/eth2-beaconchain-explorer/blob/master/templates/execution/address.html#L535C6-L535C6) with `TransactionsTable` used [here](https://github.com/gobitfly/eth2-beaconchain-explorer/blob/master/handlers/eth1Account.go#L257), which filled `fromName` by that functions results [here](https://github.com/gobitfly/eth2-beaconchain-explorer/blob/master/db/bigtable_eth1.go#L2043)). The function also calls the suggested function `GetEnsNamesForAddress` (inside of [bigtable.GetAddressNames](https://github.com/gobitfly/eth2-beaconchain-explorer/blob/master/db/bigtable_eth1.go#L1953)).

I also couldn't verify that "ZERO_ADDRESS" gets returned without db access, but that won't be the case anyways.

So it's not like the function is unrelated to ENS, but of course it also adds some overhead. The main reason I picked it is that, as mentioned, other places use it too - I thought that there might be other information than ENS (contract metadata, user/team-defined tags etc) that could be relevant, so it was used in other places. But I agree, if we just want bare ENS information, we can simply skip the overhead. Please let me know what you think of this, if you still think it should be changed I will do so.

Reply:
Good write down. I agree with your reasoning. There's always a "but" though.

BUT: The example in the screenshot uses
https://github.com/gobitfly/eth2-beaconchain-explorer/blob/7e539bd11547e1e266642f7806328358972404f2/utils/eth1.go#L107
here:
https://github.com/gobitfly/eth2-beaconchain-explorer/blob/6a5f43b1cd0c0b17e7b9679e68f64d5c22e496b7/db/bigtable_eth1.go#L2053

Why did you have to introduce a new formatting function here:
https://github.com/gobitfly/eth2-beaconchain-explorer/pull/2576/files#diff-5020779d5bcfaa28d8f25d2f4799326a80ad320c4a6f9ee0e17be6407b7c5e00R342-R347
?

I am mostly confused as we already have several places to show an ENS/address name but this function is only used for one of those. Is there potential to clean this up?

Reply:
I'm sorry for the confusion, should've explained my reasoning in more details. Since I don't want to take too much of your time, there's a TLDR and I can push another suggestion so we don't just keep discussing this back and forth.. I didn't want this to become that big, after all this is just a small issue :sweat_smile: 

**TLDR**
Essentially because a) I didn't like the _registered html-template formatting functions_ available in that case for various reasons and/or b) storing HTML-formatted data returned by `FormatAddress` among db query results [here](https://github.com/gobitfly/eth2-beaconchain-explorer/pull/2576/files#diff-aa43bc4292f439dd71618a51c3fc4eb6957ff2c8255460848da513340cc3120fR321) is bad style imo.

**Longer version**
So from my understanding, there seem to be two ways to handle frontend formatting inside of the `handlers` (relevant to this discussion at least, there could be more):
1. Some backend data is pulled from `bigtable_eth1.go`. It defines some functions `...TableData` which return formatted data. It's not necessary to apply any more formatting in the template `.html`s as a last step. E.g. [your example](https://github.com/gobitfly/eth2-beaconchain-explorer/blob/6a5f43b1cd0c0b17e7b9679e68f64d5c22e496b7/db/bigtable_eth1.go#L2053), used in html [here](https://github.com/gobitfly/eth2-beaconchain-explorer/blob/master/templates/execution/address.html#L529-L567)
2. Other data is pulled from `db.go`. It does not define functions returning data designed for frontend use. Now it's necessary to apply formatting to the results in the template `.html`s. The deposits table referenced in this PR is such a case: data set [here](https://github.com/gobitfly/eth2-beaconchain-explorer/blob/master/handlers/validator.go#L161-L180), formatting is done [around here](https://github.com/gobitfly/eth2-beaconchain-explorer/pull/2576/files#diff-62f51fd5acc30d2f4d9b8d1831f44cff0b6762c96a25b270bf3e0a7046d8e976L419). Now, the set of available formatting functions is defined [here](https://github.com/gobitfly/eth2-beaconchain-explorer/blob/master/utils/utils.go#L80-L243), and I couldn't really find one suitable for ENS names in that list (so I assumed all cases have simply been handled by case 1). There are two potentially relevant ones:
    1. `formatAddressLong`
        -  We don't currently have and don't want long addresses here if it's not an ENS
        -  Leveraging the function's early out only if a `fromName` [was set](https://github.com/gobitfly/eth2-beaconchain-explorer/pull/2576/files#diff-62f51fd5acc30d2f4d9b8d1831f44cff0b6762c96a25b270bf3e0a7046d8e976R421) (but otherwise keep `formatEth1Address` as formatting) would feel like function misuse (also it wouldn't even work if we're not actually dealing with an ENS but a custom tag or whatever got returned, see discussion above)
        - Also that function doesn't apply the href to `/address/`. Changing this behavior would obviously affect other places which I don't think is desired. And applying the href by surrounding the result with another span kind of defeats the function's purpose, as it does almost nothing more than `formatHTML` provides.
    2. `formatAddressAsLink`
        Not a bad fit actually. Unlike in i) we could even use it for the non-ens default address case for consistency reasons. The only thing I didn't like was the additional [short address after the name](https://github.com/gobitfly/eth2-beaconchain-explorer/blob/master/utils/eth1.go#L205C72-L205C80). I didn't want to affect other places again, but actually that part isn't used anyways because the function seems to be never called with a `name`.

So to sum it all up, I think there are a couple ways to solve this:
-  Define a new struct/use the generic `types.DataTableResponse` to separate backend db query data and frontend-formatted data properly, as mentioned in the initial PR comment (basically changing from method 2 to 1). That'd also be good for future changes like this, but atm it'd basically just copy the entire struct and seemed like overkill
-  Also register `utils.FormatAddress` which you linked as another template formatting function so going forward we have exactly the same behavior of ENS formatting, regardless of if method 1 or 2 is being used. Only thing I don't like too much about this is having to pass all those parameter in the html templates.. Maybe we could add another shorter version of that function (and register this one instead) which removes the `token`, `verified` (which isn't used anyways), `isContract` and `link` arguments by replacing them with default parameters
- Go with the proposed workaround solution using a new function
- Go with the existing `formatAddressAsLink` but modify it slightly (remove the short address after the ens, see above)

If we want to clean this all up, I suggest using the second option and removing `formatAddressAsLink` entirely, replacing all its uses with `utils.FormatAddress`.